src/social_network_link_prediction/similarity_methods/global_similarity/shortest_path.py

```diff
@@ -0,0 +1,17 @@
+import networkx as nx
+from scipy.sparse import csr_matrix, lil_matrix
+
+def shortest_path(G: nx.Graph, cutoff: int = None) -> csr_matrix:
+    dim = len(G.nodes)
```

Comment:
```suggestion
    dim = G.number_of_nodes()
```

Potrebbe essere più esplicativo utilizzare il metodo `number_of_nodes()`.